src/py/flwr/cli/ls.py

```diff
@@ -119,9 +136,15 @@ def ls(
             raise typer.Exit(code=1) from err
         finally:
             channel.close()
-    # pylint: disable=broad-except, unused-variable
-    except (typer.Exit, Exception):
-        _print_json_error()
+    except (typer.Exit, SystemExit, Exception) as e:  # pylint: disable=broad-except
+        if suppress_output:
+            restore_output()
+            e_message = captured_output.getvalue()
+            _print_json_error(e_message, e)
```

Comment:
Does that mean no exception message will be available in default format?

Reply:
No, if there're any exceptions in default mode, the errors are not redirected.

Reply:
But all exceptions are captured, no?

Reply:
Yes, but the error messages are not redirected. So the `if` statement doesn't affect the default mode.

Reply:
I don’t think so. All exceptions are captured here, but if you look at the code above, it only captures `ValueError`. This approach will suppress all other exceptions. For example, if there’s a gRPC exception when using the `stub` object, users will see the exception in `--format json` mode, but nothing will appear in the default mode. See below:

<img width="369" alt="Screenshot 2024-12-03 at 15 08 15" src="https://github.com/user-attachments/assets/8d059d53-8969-4a3e-b7a6-6e4040c28768">

Reply:
```suggestion
            _print_json_error(e_message, e)
        else:
            raise
```

Reply:
Resolved with `typer.secho()`.